File: WP1_data_generator_python/encapsulate.py
import numpy as np

def encapsulate(X, window):
    shape = X.shape
    length = shape[1]
    output_length = 9986
    max_valid_idx = min(length-window+1, output_length)
    
    A = np.zeros((shape[0], window, max_valid_idx), dtype=X.dtype)
    
    for i in range(max_valid_idx):
        tempWindowX = X[:, i:i+window]
        A[:, :, i] = tempWindowX
    
    return A

File: WP1_data_generator_matlab/encapsulate.m
function [A] = encapsulate(X, window)
    shape = size(X);
    length = shape (2);
   
    

    
    A = [];
 %   B = [];
    for i=1:(length-window)
        tempWindowX = X(:,i:i+(window-1));
        A = cat(3, A, tempWindowX);  %[A, tempWindowX]; 

 %       tempWindowY = Y(:,i:i+(window-1));
 %       B = cat(3, B, tempWindowY);  %[B, tempWindowY]; 
    end
end